docs/development/CodingStyle.md

```diff
@@ -0,0 +1,253 @@
+void biQuadNewLpf(float filterCutFreq, biquad_t *state, uint32_t refreshRate);
+```
+
+##Declarations
+Functions not used outside their containing .c file should be declared static (or STATIC_UNIT_TESTED so they can be used in unit tests).
+
+Non-static functions should have their declaration in a single .h file.
+
+Don't make more than necessary visible for other modules, not even types. Pre-processor macros may be used to declare module internal things that must be shared with the modules test code but otherwise hidden.
+
+In the .h file:
+```
+#ifdef MODULENAME_INTERNALS_
+… declarations …
+#endif
+```
+In the module .c file, and in the test file but nowhere else, put "#define MODULENAME_INTERNALS_" just before including the .h file.
+
+##Implementation
+Keep functions short and distinctive.
+Think about unit test when you define your functions. Ideally you should implement the test cases before implementing the function.
+
+Never put multiple statements on a single line.
+Never put multiple assignments on a single line.
+Never put multiple assignments in a single statement.
+
+Defining constants using pre-processor macros is not preferred.
+Const-correctness should be enforced.
+This allows some errors to be picked up at compile time (for example getting the order of the parameters wrong in a call to memcpy).
+
+The pre-increment is preferred over the post-increment operator, i.e.  ++i rather than i++.
+(The pre-increment form can be more efficient and it also reads better; ++i can be read as "increment i" or "inc i" whereas  i++  is read as "i plus plus").
+
+A function should only read data from the HW once in each call, and preferably all at one place.
+For example, if gyro angle or time is needed multiple times, read once and store in a local variable.
+
+Use "for" loops (rather than "do" or "while" loops) for iteration.
+
+The use of "continue" or "goto" should be avoided.
+Same for multiple "return" from a function and multiple "break" inside a "case".
+In general, they reduce readability and maintainability.
+In rare cases such constructs can be justified but only when you have considered and understood the alternatives and still have a strong reason.
+
+Use parentheses around each group in logical and mathematical statements,
+rather than relying on the implicit logic and operator priority.
+The compiler knows what it’s doing but it should be easy for people too.
+
+#Includes
+All files must include their own dependencies and not rely on includes from the included files or that some other file was included first.
+
+Do not include things you are not using.
+
+"[#pragma once](https://en.wikipedia.org/wiki/Pragma_once)" is preferred over "#include guards" to avoid multiple includes.
+
+
+#Other details
+No trailing whitespace at the end of lines or at blank lines.
+
+Stay within 80 columns, unless exceeding 80 columns significantly increases readability and does not hide information.
```

Comment:
Copied from kernel.org since you pointed in that direction.
I would be happy to remove this restriction. Because of all these very long variable names and because most of us today have big screens anyway.